### PR TITLE
Introducing compressed shadow stack instructions

### PR DESCRIPTION
**Pull request by ved-rivos**

### CODE REVIEW

File: cfi_backward.adoc [hunk 1]
```diff
@@ -56,6 +56,25 @@ Only `x1` and `x5` encodings are supported as `rs2` for `sspush`.
 When a Zimop encoding is not supported by the Zisslpcfi extension then the
 instruction follows to its Zimop defined behavior.
 
+The extension includes 16-bit versions of the `sspush x1` and `sspopchk x5`
+instructions using the compressed Zimop encodings. The `c.sspush x1` and the
```
Comment (deepak0414) [inline cfi_backward.adoc:60]: s/compressed Zimop/Zcmop

File: cfi_backward.adoc [hunk 2]
```diff
@@ -59,9 +59,8 @@ instruction follows to its Zimop defined behavior.
 The extension includes 16-bit versions of the `sspush x1` and `sspopchk x5`
 instructions using the compressed Zimop encodings. The `c.sspush x1` and the
 `c.sspopchk x5` instructions are encoded using the `C.LUI` major opcode and
-using the `c.mop.0` and `c.mop.2` encodings defined by the Zimop extension.
-Unlike the uncompressed Zimop, the compressed Zimop are defined to not write
-`rd`.
+using the `c.mop.0` and `c.mop.2` encodings defined by the Zcmop extension.
+Unlike the uncompressed Zimop, the Zcmop are defined to not write `rd`.
```
[the remark this answers is not in the document]
Comment (ved-rivos) [reply cfi_backward.adoc]: Fixed: https://github.com/riscv/riscv-cfi/pull/86/commits/712945f42875d836603f1917bb285fe2eac9a0ca

File: cfi_backward.adoc (hunk 1 above)
Comment (ved-rivos) [reply cfi_backward.adoc]: Fixed: https://github.com/riscv/riscv-cfi/pull/86/commits/339d64241264d38895ff6dc4266753195442cbdd

File: cfi_backward.adoc [hunk 3]
```diff
@@ -56,6 +56,25 @@ Only `x1` and `x5` encodings are supported as `rs2` for `sspush`.
...
+using the `c.mop.0` and `c.mop.2` encodings defined by the Zimop extension.
+Unlike the uncompressed Zimop, the compressed Zimop are defined to not write
+`rd`.
+
+[width=100%]
+[%header, cols="6,<3,>3,<3,>3,<3,>3,1,1", grid=rows, frame=none]
+|===
+|                      |`15` | `12`|`11`  |  `7`|`6` | `2`  |`1` | `0`
+|  `c.sspush   x1`  2+^|`0110`  2+^|`000001` 2+^|`00000` 2+^|`01`
+|  `c.sspopchk x5`  2+^|`0110`  2+^|`000010` 2+^|`00000` 2+^|`01`
```
[the remark this answers is not in the document]
Comment (ved-rivos) [reply cfi_backward.adoc]: Fixed : https://github.com/riscv/riscv-cfi/pull/86/commits/eac5d50d0da88c5fb2a19d74205cbf406dbf450f

File: cfi_backward.adoc [hunk 4]
```diff
@@ -53,6 +53,27 @@ Only `x1` and `x5` encodings are supported as `rd` for `sspop`.
 Only `x1` and `x5` encodings are supported as `rs1` for `sspopchk`.
 Only `x1` and `x5` encodings are supported as `rs2` for `sspush`.
 
+When a Zimop encoding is not supported by the Zisslpcfi extension then the
+instruction follows to its Zimop defined behavior.
```
[the remark this answers is not in the document]
Comment (ved-rivos) [reply cfi_backward.adoc]: Fixed: https://github.com/riscv/riscv-cfi/pull/86/commits/a8754eb859208e3f9faff5d9a6861fac83629959

File: cfi_backward.adoc [hunk 5]
```diff
@@ -53,6 +53,27 @@ Only `x1` and `x5` encodings are supported as `rd` for `sspop`.
...
+instructions using the compressed Zimop encodings. The `c.sspush x1` and the
+`c.sspopchk x5` instructions are encoded using the `C.LUI` major opcode and
+using the `c.mop.0` and `c.mop.2` encodings defined by the Zcmop extension.
+Unlike the uncompressed Zimop, the Zcmop are defined to not write `rd`.
+
+[width=100%]
+[%header, cols="6,<3,>3,<3,>3,<3,>3,1,1", grid=rows, frame=none]
+|===
+|                      |`15` | `12`|`11`  |  `7`|`6` | `2`  |`1` | `0`
+|  `c.sspush   x1`  2+^|`0110`  2+^|`000001` 2+^|`00000` 2+^|`01`
```
[the remark this answers is not in the document]
Comment (ved-rivos) [reply cfi_backward.adoc]: Fixed: https://github.com/riscv/riscv-cfi/pull/86/commits/eac5d50d0da88c5fb2a19d74205cbf406dbf450f

**Review by kwalker27 — APPROVED** [review]

**Review by kwalker27 — APPROVED** [review]

**Review by deepak0414 — APPROVED** [review]

**Pull request merged**